src/main/java/org/openehealth/app/xdstofhir/registry/query/StoredQueryMapper.java

```diff
@@ -117,7 +117,7 @@ public static void mapPatientIdToQuery(PatientIdBasedStoredQuery query, IQuery<B
         var patientId = query.getPatientId();
 
         var identifier = DocumentReference.PATIENT
-                .hasChainedProperty(Patient.IDENTIFIER.exactly().systemAndIdentifier(
+                .hasChainedProperty("Patient", Patient.IDENTIFIER.exactly().systemAndIdentifier(
```

Comment:
Can you outline the intend here? Before the query result in "DocumentReference?patient.identifier=urn:oid:1.3|id3" and now "DocumentReference?patient:Patient.identifier=urn:oid:1.3|id3" (also visible in the failing test). But i think the original one is the correct search value.

Reply:
As discussed, you have identified the issue while testing with the [Microsoft FHIR Server](https://github.com/microsoft/fhir-server). I have also started them locally (looks like no public instance is available), without this setting, we get:

`<?xml version="1.0" encoding="utf-8"?><OperationOutcome xmlns="http://hl7.org/fhir"><id value="ef85d13d-804b-4074-87d3-3a92c19640a8" /><meta><lastUpdated value="2024-05-05T14:42:39.4336098+00:00" /></meta><issue><severity value="error" /><code value="invalid" /><diagnostics value="The reference search parameter 'patient' refers to multiple possible resource types. Please specify a type in the search expression: patient:Patient or patient:Group" /></issue></OperationOutcome>`

The strange thing here is, that "patient" in DocumentReference only support 1 resource type "Patient", so not sure why MS-FHIR is mentioning "multiple possible resource types" in the error message.

Reply:
Yes, exactly this was the problem.

Reply:
For me it slightly looks like a bug or limitation of the MS fhir server. I have raised an issue in the ms fhir server repo:  microsoft/fhir-server#3839

I would like to wait for the anwer before changing. Hapi and Firely seems to work fine even with the qualified chained search parameter. If this handling is necessary to also support the MS FHir server, certain more places are necessary to adapt.

Reply:
@unixoid : Microsoft team has answered to the issue. It is a known limitation on their side and added to their backlog.

I have shortly played around if the ResourceType can be simply added to all chained searches. Only in the GetSubmissionSetsQuery it is not easily possibe, because here, a "List.item" can reference either another List (Folder) OR a DocumentReference. Spliting into 2 queries can be and acceptable workaround.

I run the also XDS Toolkit tests with the MS FHir server, the performance was quiet impressive, but it looks like their is one more limitation that cause FindDocumentsByReferenceId to be not working. Looks like the MS Fhir server seems to lake the support for the ":identifier" modifier (https://build.fhir.org/search.html#identifiercanonical). I don't think that a workaround is possible here.

*Proposal in context of this issue:*
I will merge your pull request and extend it so that the MS Fhir server will work (except FindDocumentsByReferenceId ). So Hapi, Firely and MS Fhir server can that be used.

Reply:
Merci beaucoup!